src/services/infra/StatsService.ts

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable import/prefer-default-export */
+import StatsDClient, { StatsD } from "hot-shots"
+import _ from "lodash"
+import { ModelStatic } from "sequelize"
+
+import { config } from "@config/config"
+
+import {
+  GH_MAX_REPO_COUNT,
+  GITHUB_ORG_REPOS_ENDPOINT,
+  ISOMERPAGES_REPO_PAGE_COUNT,
+} from "@root/constants"
+import { AccessToken, Site, User } from "@root/database/models"
+
+import { genericGitHubAxiosInstance } from "../api/AxiosInstance"
+
+export class StatsService {
+  private readonly statsD: StatsD
+
+  private readonly usersRepo: ModelStatic<User>
+
+  private readonly accessTokenRepo: ModelStatic<AccessToken>
+
+  private readonly sitesRepo: ModelStatic<Site>
+
+  constructor(
+    statsDClient: StatsD,
+    usersRepo: ModelStatic<User>,
+    accessTokenRepo: ModelStatic<AccessToken>,
+    sitesRepo: ModelStatic<Site>
+  ) {
+    this.statsD = statsDClient
+    this.usersRepo = usersRepo
+    this.accessTokenRepo = accessTokenRepo
+    this.sitesRepo = sitesRepo
+  }
+
+  submitV1Count = (path: string) => {
+    this.statsD.increment("versions.v1", {
+      version: "v1",
+      path,
+    })
+  }
+
+  countDbUsers = async () => {
+    // NOTE: Track only active users.
+    const numUsers = await this.usersRepo.count({
+      where: {
+        deletedAt: null,
+      },
+    })
+
+    // NOTE: Technically our e2e user will be here also
+    // but over-counting by 1 here is acceptable.
+    this.statsD.distribution("users.email.current", numUsers, 1, {
+      version: "v2",
+    })
+  }
+
+  countGithubSites = async () => {
+    const accessToken = await this.accessTokenRepo.findOne()
+    // NOTE: Cannot submit metrics if we are unable to get said metric
+    if (!accessToken) return
+
+    const sitesArr = await Promise.all(
+      _.fill(Array(ISOMERPAGES_REPO_PAGE_COUNT), null)
+        .map((__, idx) => ({
+          per_page: GH_MAX_REPO_COUNT,
+          sort: "full_name",
+          page: idx + 1,
+        }))
+        .map((params) =>
+          genericGitHubAxiosInstance
+            .get<unknown[]>(GITHUB_ORG_REPOS_ENDPOINT, {
+              params,
+            })
+            .then(({ data }) => data)
+        )
+    )
+
+    this.statsD.distribution("sites.github.all", sitesArr.flat().length, 1)
+  }
+
+  countMigratedSites = async () => {
+    const numMigratedSites = await this.sitesRepo.count({
+      where: {
+        deletedAt: null,
+      },
+    })
+
+    this.statsD.distribution("sites.db.all", numMigratedSites, 1)
+  }
+
+  trackGithubLogins = () => {
+    this.statsD.increment("users.github.login", {
+      version: "v1",
```

Comment:
they both use the same router method! i didn't distinguish here because my assumption was that we'd want to shift off github logins entirely. 

if we do want to track the version, that's possible too but is our deprecation strategy `v1 -> remove v1 + use v2 -> remove v2` or `use github login -> remove all github login`? my understanding was that it was the latter, in which case this wouldn't be useful

Reply:
this is confusing because they use the same naming convention but this **is not** v1 vs v2 sites but v1 vs v2 endpoints. a v1 site can call a v2 endpoint (and vice versa). this is because the API versioning isn't tied to the site itself but our frontend api call.

Reply:
hmm ok to be clear: 
Would logging which sites use v1 endpoint vs which site uses v2 endpoint help decide on our depreciation strategy when depreciating v1 endpoints? If yes, I would be inclined to split them up!

Reply:
for our case, given a move from `v1/<suffix>` to `v2/<suffix>`, all that's involved is changing the FE call site to use `v2/<suffix>`. 

thereafter, we monitor the `v1/<suffix>` endpoint for a reasonable time (2 wks?) to determine if there are any residual calls and if there isn't, we delete all associated code w/ the v1 endpoint.

think logging the v2 endpoints are useful just to see if they are being called at all post api version upgrade but given that we should've already checked prior to deploying the PR, i didn't include it. **however**, if you feel like this would be useful here or for **site** migration, i can add it in!

Reply:
as discussed offline, this would be useful for site migration so i'll add it in but care needs to be taken to maek sure that this isn't used to determine if the migration is successful

Reply:
added [a60b312](https://github.com/isomerpages/isomercms-backend/pull/678/commits/a60b31201e0bab98e0b4a08b1cc20bc41df00987)